tests/testsuite/binary_name.rs

```diff
@@ -0,0 +1,338 @@
+        .run();
+
+    // Check if `cargo run` is able to execute the binary
+    p.cargo("run")
+        .masquerade_as_nightly_cargo()
+        .with_stdout("Hello, crabs!")
+        .run();
+
+    p.cargo("install").masquerade_as_nightly_cargo().run();
+
+    assert_has_installed_exe(cargo_home(), "007bar");
+
+    p.cargo("uninstall")
+        .with_stderr("[REMOVING] [ROOT]/home/.cargo/bin/007bar[EXE]")
+        .masquerade_as_nightly_cargo()
+        .run();
+
+    assert_has_not_installed_exe(cargo_home(), "007bar");
+}
+
+#[cargo_test]
+fn check_env_vars() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                cargo-features = ["different-binary-name"]
+
+                [project]
+                name =  "foo"
+                version = "0.0.1"
+
+                [[bin]]
+                name = "foo"
+                filename = "007bar"
+            "#,
+        )
+        .file(
+            "src/main.rs",
+            r#"
+                fn main() {
+                    println!("{}", option_env!("CARGO_BIN_NAME").unwrap());
+                }
+            "#,
+        )
+        .file(
+            "tests/integration.rs",
+            r#"
+                #[test]
+                fn check_env_vars2() {
+                    let value = option_env!("CARGO_BIN_EXE_007bar").expect("Could not find environment variable.");
+                    assert!(value.contains("007bar"));
+                }
+            "#
+        )
+        .build();
+
+    // Run cargo build.
+    p.cargo("build").masquerade_as_nightly_cargo().run();
+    p.cargo("run")
+        .masquerade_as_nightly_cargo()
+        .with_stdout("007bar")
+        .run();
+    p.cargo("test")
+        .masquerade_as_nightly_cargo()
+        .with_status(0)
+        .run();
+}
+
+#[cargo_test]
+#[cfg(unix)]
+fn check_msg_format_json() {
+    // Create the project.
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                cargo-features = ["different-binary-name"]
+
+                [project]
+                name =  "foo"
+                version = "0.0.1"
+
+                [[bin]]
+                name = "foo"
+                filename = "007bar"
+                path = "src/main.rs"
+            "#,
+        )
+        .file("src/main.rs", "fn main() { assert!(true) }")
+        .build();
+
+    let output = r#"
+{
+    "reason": "compiler-artifact",
+    "package_id": "foo 0.0.1 [..]",
+    "manifest_path": "[CWD]/Cargo.toml",
+    "target": "{...}",
+    "profile": "{...}",
+    "features": [],
+    "filenames": "{...}",
+    "executable": "[ROOT]/foo/target/debug/007bar[EXE]",
+    "fresh": false
+}
+
+{"reason":"build-finished", "success":true}
+"#;
+
+    // Run cargo build.
+    p.cargo("build --message-format=json")
+        .masquerade_as_nightly_cargo()
+        .with_json(output)
+        .run();
+}
+
+#[cargo_test]
+#[cfg(windows)]
+fn check_msg_format_json() {
```

Comment:
I think this entire test can be removed now that the other supports both platforms?